packages/imt.sol/contracts/internal/InternalLeanIMT.sol

```diff
@@ -46,14 +46,20 @@ library InternalLeanIMT {
             revert LeafAlreadyExists();
         }
 
-        while (2 ** self.depth < self.size + 1) {
+        // Cache tree size to optimize gas
+        uint256 treeSize = self.size;
+
+        while (2 ** self.depth < treeSize + 1) {
             self.depth += 1;
         }
 
-        uint256 index = self.size;
+        // Cache tree depth to optimize gas
+        uint256 treeDepth = self.depth;
+
+        uint256 index = treeSize;
```

Comment:
Can we use `index` directly instead of `treeSize` at line 50? That should save an assignment

Reply:
Yes, makes sense.